src/openforms/registrations/contrib/zgw_apis/plugin.py

```diff
@@ -464,6 +548,55 @@ def register_submission(self, submission: Submission, options: dict) -> dict | N
                 "intermediate.objects_api_zaakobject",
             )
 
+        # Map variables to case eigenschappen (if a connection has been used in ZGW registration)
+        if variables_properties := options.get("property_mappings"):
+            property_mappings = get_property_mappings_from_submission(
+                submission, variables_properties
+            )
+
+            eigenschappen = execute_unless_result_exists(
+                partial(catalogi_client.list_eigenschappen, options["zaaktype"]),
+                submission,
+                "intermediate.zaaktype_eigenschappen",
+            )
+
+            retrieved_eigenschappen = {
+                eigenschap["naam"]: {
+                    "url": eigenschap["url"],
+                    "specificatie": eigenschap["specificatie"],
+                }
+                for eigenschap in eigenschappen
+            }
+
+            for key, value in property_mappings.items():
+                if key in retrieved_eigenschappen:
+                    processed_value = process_according_to_eigenschap_format(
+                        retrieved_eigenschappen[key].get("specificatie"), value
```

Comment:
```suggestion
                        retrieved_eigenschappen[key]["specificatie"], value
```
your dict comprehension above already guarantees that the key exists